build: update stemmer to version 2.0.0

gkalpak opened the pull request:
NOTE:
`stemmer` v2.0.0 switched to ES modules (see words/stemmer@03519229c84555a38d3256409d1871edfd43528f), which means that the only way to consume it in our CommonJS  setup (for example, in [generateKeywords][1]) is via an async `import()`.

This commit makes the `generateKeywords` processor asynchronous in order to be able to dynamically import and use `stemmer`.

[1]: https://github.com/angular/angular/blob/251bec159af1e8f90c3af9820695c645800207e4/aio/tools/transforms/angular-base-package/processors/generateKeywords.js

mary-poppins commented:
You can preview f7208ad at https://pr41724-f7208ad.ngbuilds.io/.

petebacondarwin reviewed (commented):
I know this is still draft, but I feel like it would be nicer to just pass the stemmer (either via a parameter or via a closure) to the `tokenize()` function and avoid making all of those `async`.

mary-poppins commented:
You can preview 49e1e88 at https://pr41724-49e1e88.ngbuilds.io/.

gkalpak committed:
build: update stemmer to version 2.0.0

NOTE:
`stemmer` v2.0.0 switched to ES modules (see
words/stemmer@03519229c84555a38d3256409d1871edfd43528f), which means
that the only way to consume it in our CommonJS setup (for example, in
[generateKeywords][1]) is via an async `import()`.

This commit makes the `generateKeywords` processor asynchronous in order
to be able to dynamically import and use `stemmer`.

[1]: https://github.com/angular/angular/blob/251bec159af1e8f90c3af9820695c645800207e4/aio/tools/transforms/angular-base-package/processors/generateKeywords.js

mary-poppins commented:
You can preview 67ac096 at https://pr41724-67ac096.ngbuilds.io/.

petebacondarwin reviewed (approved):
This works for now, but I think we need to come up with a more general solution as more packages move over to native ES modules.

mary-poppins commented:
You can preview fd003cc at https://pr41724-fd003cc.ngbuilds.io/.

jessicajaniuk commented:
@gkalpak This PR has been merged into `master` and `12.0.x`, but has conflicts with `11.2.x`. Please create a patch PR to get it added to `11.2.x`.

gkalpak commented:
For reference, here is the PR for patch: #41854
